pywbem/_logging.py

```diff
@@ -14,404 +14,341 @@
+* ``pywbem.api`` - Logs user-issued calls to :class:`~pywbem.WBEMConnection`
+  methods that drive WBEM operations (see :ref:`WBEM operations`) and their
+  responses before they are passed back to the user. Log entries are defined at
+  DEBUG log level. This logs the parameters of each request and the results
+  including CIMObjects/exceptions in each method response. It also logs the
+  creation of the :class:`~pywbem.WBEMConnection` object to capture connection
+  information.
+
+* ``pywbem.http`` - Logs HTTP requests and responses between the pywbem client
+  and WBEM server. This logs the http request data and response data including
+  http headers and XML payload.
+
+The above logger names output log records when they are configured and
+activated. All pywbem loggers  log at the DEBUG log level and
+logging is activated for one or more :class:`~pywbem.WBEMConnection` instance
+objects.
+
+There are three steps to configuration of the pywbem loggers:
+
+* **Configure Python logger names** - Sets the Python logger parameters
+  (logging handlers, message formats, etc.). Either Python logging
+  methods or methods defined in :ref:`Logging configuration methods` may be
+  used to configure the loggers. The `dest` parameter of the pywbem methods
+  configures the loggers for either stream or file loggers and the `name`
+  parameter defines which of the pywbem loggers is being configured. This
+  configure step does not activate any pywbem logger. If no loggers are
+  configured, pywbem defaults to a logger with the Python
+  logging NullHandler.
+
+* **Set the log record detail level for pywbem loggers(Optional)** - Because
+  pywbem logs can generate large quantities of information, they allow the user
+  to control the quantity of information in each log record.  This is a pywbem
+  feature so it can only be configured using the pywbem methods defined in
+  :ref:`Logging configuration methods` using the `detail_level` parameter. The
+  `detail_level` possible values are defined in
+  :data:`~pywbem_logging.LOG_DETAIL_LEVELS`. The default for `detail_level` is
+  `all` which logs all if the data in the requests and responses if the
+  `detail_level` parameter is not set.
+
+* **Activate the pywbem loggers** - A pywbem logger is activated when it is
+  connected to one or more WBEMConnections so that log records will be generated
+  when any of the WBEMConnection methods that communicates with a WBEM server
+  is executed for that WBEMConnection. Pywbem separates the configuration
+  of the loggers from activation to allow logger configuration by either pywbem
+  methods or standard Python logging methods. Activation is controlled only by
+  the `connection` parameter of  the methods defined in
+  :ref:`Logging configuration methods` which either activates logging for
+  all (`connect=True`) subsequently created :class:`~pywbem.WBEMConnection`
+  objects or for a particular connection (`connect=conn` where conn is a
+  WBEMConnection object).
```

Comment:
Multiple comments on this paragraph:

* On the first sentence: It seems this is the only remaining place where we phrase the relation it using "connected to". Also, the notion of "attached to" no longer seems to be used. I think this is all goodness since I found it difficult to understand. I suggest that we also get rid of this last "connected to". Also, the first sentence is generally a bit too complicated. I suggest removing the first sentence altogether. The third sentence is really sufficient to explain it at this level.

* The second sentence motivates the existence of a logger activation concept, but it states a fact that is true but is not the reason we introduced the activation concept. The real reason it was introduced is that we want to save performance by allowing loggers to be deactivated when they are not used. Suggest changing that motivation statement accordingly.

* In the third sentence "Activation is controlled", I don't think we need to describe how it is done, just that there are these two possibilities.

Reply:
Simplified the first sentence and removed use of connect to.

I just removed second sentence.  It was not really adding much